src/components/AdminSettings.vue

```diff
@@ -350,11 +597,80 @@ export default {
 			this.serverHostUrlForEdit = this.state.openproject_instance_url
 			this.isOpenProjectInstanceValid = null
 		},
-		setNCOAuthFormToViewMode() {
+		setGroupFolderSetUpToEditMode() {
+			this.formMode.groupFolderSetUp = F_MODES.EDIT
+			this.isFormCompleted.groupFolderSetUp = false
+			this.isProjectFolderSwitchEnabled = this.currentGroupFolderState
+		},
+		setManagedGroupFolderSetupToViewMode() {
+			this.currentGroupFolderState = true
+			this.textLabelGroupFolderSetupButton = t('integration_openproject', 'Keep current change')
+			this.isFormCompleted.groupFolderSetUp = true
+			this.formMode.groupFolderSetUp = F_MODES.VIEW
+			this.isGroupFolderSetupCorrect = true
+		},
+		async setNCOAuthFormToViewMode() {
 			this.formMode.ncOauth = F_MODES.VIEW
 			this.isFormCompleted.ncOauth = true
+			if (!this.isIntegrationComplete) {
+				this.showDefaultManagedFolders = true
+				this.isProjectFolderSwitchEnabled = true
+				this.textLabelGroupFolderSetupButton = t('integration_openproject', 'Setup OpenProject user, group and folder')
```

Comment:
I think we should store the texts used in buttons like  t('integration_openproject', 'Setup OpenProject user, group and folder') in a constant variable  with some appropriate and then use that variable everywhere needed, I see a lot of repetitions of the text
```suggestion
				this.textLabelGroupFolderSetupButton = textForSettingUpGroupFolders
```